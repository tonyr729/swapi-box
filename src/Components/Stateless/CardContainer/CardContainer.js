const CardContainer = (props) => {
  

  return(
    <div>

    </div>
  )
}