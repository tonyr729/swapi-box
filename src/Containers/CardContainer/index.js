import React from 'react';
import Card from '../../Components/Stateless/Card/';
import './index.css';
import PropTypes from 'prop-types';

const CardContainer = ({data, setFavorites, favorites}) => {
  const cards = data.displayed.map((personObject, index)=> {
    return <Card key={index} data={personObject} setFavorites={setFavorites} favorites={ favorites }/>
  })

  return(
    <div className="card-container">
      {cards}
    </div>
  )
}

CardContainer.propTypes = {
  data: PropTypes.object
};

export default CardContainer;