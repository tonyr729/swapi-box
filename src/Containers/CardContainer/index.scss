.card-container {
  display: flex;
  flex-wrap: wrap;
  height: 45rem;
  overflow: scroll;
}