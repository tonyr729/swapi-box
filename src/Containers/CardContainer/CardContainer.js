import React from 'react';
import Card from '../../Components/Stateless/Card/Card'
import './CardContainer.css'

const CardContainer = ({data, setFavorites}) => {
  const cards = data.displayed.map((personObject, index)=> {
    return <Card key={index} data={personObject} setFavorites={setFavorites} />
  })

  return(
    <div className="card-container">
      {cards}
    </div>
  )
}

export default CardContainer;