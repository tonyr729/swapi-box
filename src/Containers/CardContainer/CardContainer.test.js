import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CardContainer from './CardContainer'

describe('CardContainer', () => {

  describe('Component', () => {
    it('should match snapshot', () => {
      const cardContainer = renderer.create(<CardContainer data={[{name: "tony"}]}/>).toJSON();

      expect(cardContainer).toMatchSnapshot();
    })
  })
})