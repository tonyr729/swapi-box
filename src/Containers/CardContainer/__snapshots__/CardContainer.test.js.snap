// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer Component should match snapshot 1`] = `
<div
  className="card-container"
>
  <div
    className="card"
  >
    <p>
      0
      : 
      t
    </p>
    <p>
      1
      : 
      o
    </p>
    <p>
      2
      : 
      n
    </p>
    <p>
      3
      : 
      y
    </p>
    <button
      onClick={[Function]}
    >
      â˜…
    </button>
  </div>
</div>
`;
